File: Ejemplos/Procesos/C/fork_wait.c
#include <stdlib.h>
#include <stdio.h>
#include <unistd.h>
#include <sys/wait.h>

int main(int argc, char *argv[])
{
    pid_t pid = fork();

    if( pid < 0 )
    {
        printf("Error de creación\n");
        return EXIT_FAILURE;
    }
    
    if( pid )
    {
        printf("Soy el padre. PID: %d\n",getpid());
        wait(NULL);
    }
    else
    {
        printf("Soy el hijo. PID: %d\n",getpid());
        printf("Mi padre es PID: %d\n",getppid());
        return EXIT_SUCCESS;	
    }

    return EXIT_SUCCESS;
}

File: Ejemplos/Procesos/Python/fork_wait.py
import os
import sys

pid = os.fork()

if pid < 0:
    sys.exit("Error al crear el nuevo proceso")

if pid:
    print("Soy el proceso padre. PID: ", os.getpid() )
    print( os.wait() )
else:
    print("Soy el proceso hijo. PID: ",  os.getpid())
    print("Mi padre es el proceso con PID: ",  os.getppid())
    os._exit( os.EX_OK )
